gradle-baseline-java/src/main/groovy/com/palantir/baseline/tasks/CheckJUnitDependencies.java

```diff
@@ -63,6 +71,17 @@ public final void validateDependencies() {
                 });
     }
 
+    @Classpath
+    public final Provider<List<Configuration>> getConfigurations() {
```

Comment:
I think gradle will get angry at you for using Provider<List<>> instead of ListProvider. If you want to have dynamic set of inputs you have to do it via `getInputs()` similarly to how you handle outputs

Reply:
this is using the same pattern as check unused dependencies. Propeties are differnet to providers, right?

Reply:
I think this should be fine, right? There was a bug in Gradle 6.something that tried to cast `Provider`s to `CollectionPropertyInternal` at some point, forcing you to use `Property`s, but this was fixed?